tests/test_interfaces.py

```diff
@@ -8,7 +8,33 @@
 
 import six
 
-from cryptography.utils import InterfaceNotImplemented, verify_interface
+from cryptography.utils import (
+    InterfaceNotImplemented, register_interface_if, verify_interface
+)
+
+
+def test_register_interface_if_true():
+    @six.add_metaclass(abc.ABCMeta)
+    class SimpleInterface(object):
+        pass
+
+    @register_interface_if(1 == 1, SimpleInterface)
+    class SimpleClass(object):
+        pass
+
+    assert issubclass(SimpleClass, SimpleInterface) is True
```

Comment:
No, the asserts rae equally meaningful, `issubclass(SimpleClass,
SimpleInterface) == isinstance(SimpleClass(), SimpleInterface)`

On Mon, Aug 29, 2016 at 7:57 AM, Paul Kehrer <notifications@github.com>
wrote:

> In tests/test_interfaces.py
> <https://github.com/pyca/cryptography/pull/3120#discussion_r76594680>:
>
> > -from cryptography.utils import InterfaceNotImplemented, verify_interface
> > +from cryptography.utils import (
> > +    InterfaceNotImplemented, register_interface_if, verify_interface
> > +)
> > +
> > +
> > +def test_register_interface_if_true():
> > +    @six.add_metaclass(abc.ABCMeta)
> > +    class SimpleInterface(object):
> > +        pass
> > +
> > +    @register_interface_if(1 == 1, SimpleInterface)
> > +    class SimpleClass(object):
> > +        pass
> > +
> > +    assert issubclass(SimpleClass, SimpleInterface) is True
>
> Why issubclass and not isinstance here? It probably doesn't matter but I'm
> curious if there's any effective difference for our purposes
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/pyca/cryptography/pull/3120/files/ca4c98d05fd8b8a239e6a7a21e14a4f9c5df428d#r76594680>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAADBLtA3JUxowtCsOoRZ2FEB1slfNFlks5qksk8gaJpZM4JvS_Z>
> .
>



-- 
"I disapprove of what you say, but I will defend to the death your right to
say it." -- Evelyn Beatrice Hall (summarizing Voltaire)
"The people's good is the highest law." -- Cicero
GPG Key fingerprint: D1B3 ADC0 E023 8CA6